loader.go

```diff
@@ -261,43 +261,44 @@ func (l *Loader) attachBpfProgs() error {
 	return err
 }
 
-func (l *Loader) EventLoop(ctx context.Context, out chan<- Event, errs chan<- error) {
-	in := make(chan ringbuf.Record, l.objs.bpfMaps.Ringbuf.MaxEntries())
-
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			default:
-				record, err := l.reader.Read()
-				if errors.Is(err, ringbuf.ErrClosed) {
-					break
-				}
-				if err != nil {
-					errs <- err
-					continue
-				}
-				in <- record
-			}
-		}
-	}()
+type Record struct {
+	Event *Event
+	Error error
+}
 
+func (l *Loader) EventLoop(ctx context.Context, out chan<- Record) {
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case record := <-in:
-			event, err := NewEvent(record.RawSample)
+		default:
+			var r Record
+
+			record, err := l.reader.Read()
```

Comment:
here `Read` ends up utilising an epoll wait with a timeout -1 if the reader deadline is not set, which I think we do not set. So just checking here, we are okay ctx.Done being essentially ignored when we get to Read as this is blocking until either an event is produced or something closing the reader?

Reply:
yeah in theory it's fine, a running system will always generate events. I'm open to alternatives though

Reply:
pushed some changes, let me know if it looks good